pythonAPIClient/client.py

```diff
@@ -297,9 +297,148 @@ def usage_by_client(self, clientID, owner):
+                    However, video_path can exist even if 'is_active_video'
+                    is false for this ip camera.
+
+                Raises
+                ------
+                    AuthenticationError
+                        If the client secret of this client object does not match the clientID.
+                    FormatError
+                        List of invalid attributes.
+                    ResourceConflictError
+                        The legacy_cameraID already exist in the database.
+                    InternalError
+                        If there is an API internal error.
+                    ResourceNotFoundError
+                        If no client app exists with the clientID of this client object.
+
+                Returns
+                -------
+                str
+                    cameraID: the new camera ID for the created camera
+        """
+
+        local_params = dict(locals())
+
+        if self.token is None:
+            self.request_token()
+
+        url = Client.base_URL + 'cameras/create'
+
+        local_params['type'] = local_params.pop('camera_type')
+        del local_params['self']
+
+        if camera_type == 'ip':
+            if ip is not None:
+                local_params['retrieval'] = {
+                    'ip': local_params.pop('ip'),
+                    'port': local_params.pop('port'),
+                    'brand': local_params.pop('brand'),
+                    'model': local_params.pop('model'),
+                    'image_path': local_params.pop('image_path'),
+                    'video_path': local_params.pop('video_path')
+                }
+                local_params['retrieval'] = json.dumps(local_params['retrieval'])
+        elif camera_type == 'non-ip':
+            if snapshot_url is not None:
+                local_params['retrieval'] = {
+                    'snapshot_url': local_params.pop('snapshot_url')
+                }
+                local_params['retrieval'] = json.dumps(local_params['retrieval'])
+        elif camera_type == 'stream':
+            if m3u8_url is not None:
+                local_params['retrieval'] = {
+                    'm3u8_url': local_params.pop('m3u8_url')
+                }
+                local_params['retrieval'] = json.dumps(local_params['retrieval'])
+
+        response = self._check_token(requests.post(url, data=local_params,
+                                                   headers=self.header_builder()), flag='POST',
+                                     url=url, data=local_params)
+        if response.status_code != 201:
+            if response.status_code == 403:
+                raise AuthenticationError(response.json()['message'])
+            elif response.status_code == 422:
+                raise FormatError(response.json()['message'])
+            elif response.status_code == 409:
+                raise ResourceConflictError(response.json()['message'])
+            elif response.status_code == 404:
```

Comment:
We are already checking 404 in request_token, therefore we don't have to check it here.